### PR TITLE
Integrate deal making with Filecoin via Singularity

### LINKED ISSUE #113: S3 connector update for milestone 2 (run the whole docker)
Update the docker file. https://github.com/filecoin-project/motion-s3-connector

### PR DESCRIPTION
**Pull request by masih**
Integrate start-up with Singularity and configuration to make deals with Filecoin storage providers.

Update README, and add example env files.

Fixes: https://github.com/filecoin-project/motion/issues/113

### CODE REVIEW

**Comment by masih** [comment]
@willscott @hannahhoward thoughts on what to do with the CI test in this repo? Do we want to set up devnet here again or should we just disable it?

**Comment by willscott** [comment]
I think having the integration CI running in the motion repo is the right place - that's where more changes will be and where we want to verify that the changes don't break s3 functionality.

File: docker-compose.yaml [hunk 1]
```diff
@@ -1,14 +1,127 @@
...
+      DATABASE_CONNECTION_STRING: postgres://${SINGULARITY_DB_USER:-postgres}:${SINGULARITY_DB_PASSWORD:-postgres}@db:5432/${SINGULARITY_DB_NAME:-singularity}
+      LOTUS_TEST:
+      LOTUS_API:
+      LOTUS_TOKEN:
+    depends_on:
+      db:
+        condition: service_healthy
+
+  singularity_content_provider:
+    image: ghcr.io/data-preservation-programs/singularity:${SINGULARITY_VERSION:-0.3.2}
```
Comment (xinaxu) [inline docker-compose.yaml:80]: Is it the right version

**Review by xinaxu — APPROVED** [review]

**Review by gammazero — APPROVED** [review]

File: docker-compose.yaml (hunk 1 above)
Comment (masih) [reply docker-compose.yaml:80]: I think it is the one that works with Motion 0.2.0?

Comment (masih) [reply docker-compose.yaml:80]: Going ahead with this version until there is a new tagged version of motion that's tested with singularity.

Comment (masih) [reply docker-compose.yaml:80]: Version to be updated once [this](https://github.com/filecoin-project/motion/issues/147) is sorted out. Cc @gammazero

**Pull request merged**